cmd/keymasterd/config.go

```diff
@@ -423,28 +424,40 @@ func generateArmoredEncryptedCAPritaveKey(passphrase []byte, filepath string) er
 	if err != nil {
 		return err
 	}
-
 	encryptionType := "PGP MESSAGE"
 	armoredBuf := new(bytes.Buffer)
 	armoredWriter, err := armor.Encode(armoredBuf, encryptionType, nil)
 	if err != nil {
 		return err
 	}
-
-	plaintextWriter, err := openpgp.SymmetricallyEncrypt(armoredWriter, passphrase, nil, nil)
+	var plaintextWriter io.WriteCloser
+	if len(passphrase) < 1 {
+		plaintextWriter, err = os.OpenFile(filepath, os.O_CREATE|os.O_WRONLY,
+			0600)
+	} else {
+		plaintextWriter, err = openpgp.SymmetricallyEncrypt(armoredWriter,
+			passphrase, nil, nil)
+	}
 	if err != nil {
 		return err
 	}
-
-	privateKeyPEM := &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(privateKey)}
+	privateKeyPEM := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(privateKey),
+	}
 	if err := pem.Encode(plaintextWriter, privateKeyPEM); err != nil {
 		return err
 	}
-	plaintextWriter.Close()
-	armoredWriter.Close()
-
-	//os.Remove(filepath)
-	return ioutil.WriteFile(filepath, armoredBuf.Bytes(), 0600)
+	if err := plaintextWriter.Close(); err != nil {
+		return err
+	}
+	if len(passphrase) < 1 {
+		return nil
+	} else {
+		plaintextWriter.Close()
```

Comment:
you arlready closed the plaintext File here, no need for this line

Reply:
Deleted.